src/components/Login/Login.js

```diff
@@ -140,6 +140,7 @@ class Login extends Component {
   }
 
   signInFB() {
+    this.setState({ loaded: false });
```

Comment:
(If I understood your question correctly ...)
I think that all procedures are performed sequentially because next procedure starts after finishing 'setState' procedure.